core/interoperation/src/lib.rs

```diff
@@ -93,15 +101,25 @@ impl Interoperation for InteroperationImpl {
         // syscalls3 are enabled. Due to only the hardfork field in consensus and the
         // epoch field in tx_env is used, the provided arguments only need to fill these
         // fields correctly.
-        let (ckb_spec, ckb2023_disabled_env) = (
-            Arc::new(Consensus::default()),
-            Arc::new(TxVerifyEnv::new_commit(
-                &HeaderBuilder::default()
-                    .number(CKB2023_DISABLED_NUMBER.pack())
-                    .epoch(CKB2023_DISABLED_EPOCH.pack())
-                    .build(),
-            )),
-        );
+        let (ckb_spec, ckb2023_disabled_env) = {
+            let env = match version {
+                CKBVMVersion::V2021 => TxVerifyEnv::new_commit(
+                    &HeaderBuilder::default()
+                        .number(CKB2023_DISABLED_NUMBER.pack())
+                        .epoch(CKB2023_DISABLED_EPOCH.pack())
+                        .build(),
+                ),
+                // TODO: Determine 2023 activation time
+                CKBVMVersion::V2023 => TxVerifyEnv::new_commit(
+                    &HeaderBuilder::default()
+                        .number(CKB2023_DISABLED_NUMBER.pack())
+                        .epoch(CKB2023_DISABLED_EPOCH.pack())
+                        .build(),
+                ),
+            };
+
+            (Arc::new(Consensus::default()), Arc::new(env))
```

Comment:
Consensus includes hardfork_switch so it should be different depending on hardfork?

Reply:
no，it depend on tx_env https://github.com/nervosnetwork/ckb/blob/develop/script/src/verify.rs#L544-L548

Reply:
Doesn't this depend on both `consensus.hardfork_switch()` and `tx_env`?

Reply:
`consensus.hardfork_switch()` confirms the activated epoch, this is a fixed value
`tx_env` confirms the epoch the transaction is in. It belongs to the new environment if it is greater than the epoch in `hardfork_switch`.

Reply:
OK this works because `Consensus::default()` uses mirana hardfork switches. But it's kind of undocumented. I'm not sure whether it'll change in later versions of ckb.